features/cloud.py

```diff
@@ -481,6 +496,97 @@ def _create_instance(
         return inst
 
 
+class GCE(Cloud):
+    name = "gce"
+
+    """Class that represents the Azure cloud provider.
+
+    :param gce_credentials_path
+        The GCE credentials path to use when authentiacting to GCE
+    :param gce_project
+        The name of the GCE project to be used
+    :machine_type:
+        A string representing the type of machine to launch (pro or generic)
+    :region:
+        The region to create the resources on
+    :tag:
+        A tag to be used when creating the resources on the cloud provider
+    :timestamp_suffix:
+        Boolean set true to direct pycloudlib to append a timestamp to the end
+        of the provided tag.
+    """
+
+    env_vars: "Tuple[str, ...]" = ("gce_credentials_path", "gce_project")
+
+    def __init__(
+        self,
+        machine_type: str,
+        region: "Optional[str]" = "us-west2",
+        tag: "Optional[str]" = None,
+        timestamp_suffix: bool = True,
+        zone: "Optional[str]" = "a",
+        gce_credentials_path: "Optional[str]" = None,
+        gce_project: "Optional[str]" = None,
+    ) -> None:
+        self.gce_credentials_path = gce_credentials_path
+        self.gce_project = gce_project
+        self.zone = zone
+        os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = str(
```

Comment:
Let's also set os.environ["GOOGLE_CLOUD_PROJECT"] = str(self.gce_project)

without that I get 
WARNING:google.auth._default:No project ID could be determined. Consider running `gcloud config set project` or setting the GOOGLE_CLOUD_PROJECT environment variable

Reply:
We are passing that variable through the `UACLIENT_BEHAVE_GCP_PROJECT`` env var. It is weird that this warning is popping up, since the pycloudlib implementation actually requires the project named to be passed explicitly, not though an env var